Task 4/Task4/Task4/CastomGit.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Actions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Task4
+{
+    class CastomGit
+    {
+        private string path;
+
+        private List<(DateTime, FileAction)> Log = new List <(DateTime, FileAction)>();
+
+        public Dictionary<String, String> FilePastPath = new Dictionary<string, string>();
+
+        private UserCommunication communicator = new UserCommunication();
+
+        private DateTime LogTime;
+
+        public CastomGit(string path)
+        {
+            this.path = path;
+        }
+
+        
+        public void StartProgram()
+        {
+            string[] files = Directory.GetFiles(path, "*.txt");
+
+            DateTime currentTime = DateTime.Now;
+
+            string initialCommitPath = path + "\\" + currentTime.ToString("dd.MM.yyyy.HH.mm.ss");
+
+            Directory.CreateDirectory(initialCommitPath);
+
+            for (int i = 0; i< files.Length; i++)
+            {
+                String text = File.ReadAllText(files[i]);
+
+                File.WriteAllText(initialCommitPath + files[i].Remove(0, path.Length), text);
+
+                FilePastPath.Add(files[i], initialCommitPath + files[i].Remove(0, path.Length));
+            }
+
+            Tracking();
+
+        }
+
+
+        public void Tracking()
```

Comment:
Methods should contain verbs in 1st form, like an instruction point "what should be done next? Oh, it says `StartTracking`"